### PR TITLE
ddl: support scatter region when create table with split single region

### PR DESCRIPTION
**Pull request by River2000i**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/55184 ref https://github.com/tikv/pd/issues/8424

Problem Summary: Enable `split-table`, create table without other options(such as partitions, `pre_split_regions` ), TiKV only will split one region. If enable `tidb_scatter_region`, PD will compare the region count record by `engineContext` in https://github.com/tikv/pd/blob/13174b5d4cab4d958f0b4ea2718ea7bb74992bc7/pkg/schedule/scatter/region_scatterer.go#L444. 

Since in first time call scatter region, all region count is 0 in `engineContext`. So it will pick the origin store. IOW, `tidb_scatter_region` doesn't work when there is only one region in table/group level. For these table, scatter region in cluster level.

### What changed and how does it work?
1. Set `tableID` to nil. All new table with only one region will be the same group in PD. Add  `scatterRegionByClusterLevel` in `SplitRecordRegion()` decide `tableID` is nil or not. 
2. Add `scatter` in `preSplitAndScatter()`

since the pr base on PD schedule(include RPC server and scatter region), UT is hard to cover. Add test in PD is better.
manual test result:
create table in 10 sessions parallel
image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by River2000i** [comment]
Issue Number: ref https://github.com/tikv/pd/issues/8424

**Comment by River2000i** [comment]
@HuSharp @lance6716 @Benjamin2037 PTAL~

File: pkg/ddl/table.go [hunk 1]
```diff
@@ -870,7 +870,14 @@ func (w *worker) onTruncateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver i
...
 		partitions = tblInfo.GetPartitionInfo().Definitions
 	}
-	preSplitAndScatter(w.sess.Context, d.store, tblInfo, partitions)
+	var scatterRegion bool
+	val, err := w.sess.Context.GetSessionVars().GetGlobalSystemVar(context.Background(), variable.TiDBScatterRegion)
+	if err != nil {
+		logutil.DDLLogger().Warn("won't scatter region", zap.Error(err))
+	} else {
+		scatterRegion = variable.TiDBOptOn(val)
+	}
```
Comment (D3Hunter) [inline pkg/ddl/table.go:879]: duplicate many times

**Comment by D3Hunter** [comment]
need discuss, and PM to define the behavior

**Pull request closed without merging**